### PR TITLE
refactor(target_chains/ethereum): remove batch update support

### PR DESCRIPTION
**Pull request by Riateche**
- Removed support for old batch update format from `updatePriceFeeds`, `parsePriceFeedUpdates`, `getUpdateFee` contract functions.
- Updated and cleaned up tests accordingly.

Batch renames are made in separate commits for easier review.

`PythUpgradable` contract size reduced from 22.061 kB to 20.568 kB, according to `forge build --sizes`.

### Gas report

No significant changes, according to `forge test --gas-report --match-contract GasBenchmark`.
<details><summary>Reports</summary>

Old:

| contracts/pyth/PythUpgradable.sol:PythUpgradable contract |                 |        |        |        |         |
|-----------------------------------------------------------|-----------------|--------|--------|--------|---------|
| Deployment Cost                                           | Deployment Size |        |        |        |         |
| 4848557                                                   | 22459           |        |        |        |         |
| Function Name                                             | min             | avg    | median | max    | # calls |
| getEmaPrice                                               | 4025            | 4025   | 4025   | 4025   | 1       |
| getPrice                                                  | 3169            | 3169   | 3169   | 3169   | 1       |
| getUpdateFee                                              | 2455            | 3370   | 3553   | 3553   | 426     |
| initialize                                                | 245301          | 245301 | 245301 | 245301 | 35      |
| parsePriceFeedUpdates                                     | 118792          | 151982 | 143161 | 207422 | 10      |
| parsePriceFeedUpdatesUnique                               | 121266          | 122119 | 122119 | 122973 | 2       |
| updatePriceFeeds                                          | 103293          | 279601 | 328963 | 328963 | 47      |
| updatePriceFeedsIfNecessary                               | 7478            | 87091  | 87091  | 166705 | 2       |

New:

| contracts/pyth/PythUpgradable.sol:PythUpgradable contract |                 |        |        |        |         |
|-----------------------------------------------------------|-----------------|--------|--------|--------|---------|
| Deployment Cost                                           | Deployment Size |        |        |        |         |
| 4526025                                                   | 20966           |        |        |        |         |
| Function Name                                             | min             | avg    | median | max    | # calls |
| getEmaPrice                                               | 4025            | 4025   | 4025   | 4025   | 1       |
| getPrice                                                  | 3169            | 3169   | 3169   | 3169   | 1       |
| getUpdateFee                                              | 3553            | 3553   | 3553   | 3553   | 315     |
| initialize                                                | 245301          | 245301 | 245301 | 245301 | 31      |
| parsePriceFeedUpdates                                     | 121062          | 173233 | 188705 | 207350 | 10      |
| parsePriceFeedUpdatesUnique                               | 121242          | 122095 | 122095 | 122949 | 2       |
| updatePriceFeeds                                          | 102690          | 300067 | 353387 | 353387 | 41      |
| updatePriceFeedsIfNecessary                               | 7478            | 99299  | 99299  | 191120 | 2       |
</details>

### CODE REVIEW

File: target_chains/ethereum/contracts/forge-test/VerificationExperiments.t.sol [hunk 1]
```diff
@@ -483,6 +499,121 @@ contract PythExperimental is Pyth {
         updateLatestPriceIfNecessary(priceId, info);
     }
 
+    function parseSingleAttestationFromBatch(
```
Comment (ali-bahjati) [inline target_chains/ethereum/contracts/forge-test/VerificationExperiments.t.sol:502]: I think it's fine to just remove it entirely from here.

File: target_chains/ethereum/contracts/forge-test/Pyth.t.sol [hunk 2]
```diff
@@ -12,156 +12,123 @@ import "./utils/WormholeTestUtils.t.sol";
 import "./utils/PythTestUtils.t.sol";
 import "./utils/RandTestUtils.t.sol";
 
-contract PythTest is Test, WormholeTestUtils, PythTestUtils, RandTestUtils {
+contract PythTest is Test, WormholeTestUtils, PythTestUtils {
```
Comment (ali-bahjati) [inline target_chains/ethereum/contracts/forge-test/Pyth.t.sol:15]: This is great that you've refactored the tests here to be with Accumulator. But I'm wondering whether they were already being tested in Pyth.WormholeMerkleAccumulator.t.sol. If they are already covered I think we can just remove the tests here.

File: target_chains/ethereum/sdk/solidity/IPythEvents.sol [hunk 3]
```diff
@@ -15,9 +15,4 @@ interface IPythEvents {
         int64 price,
         uint64 conf
     );
-
-    /// @dev Emitted when a batch price update is processed successfully.
-    /// @param chainId ID of the source chain that the batch price update comes from.
-    /// @param sequenceNumber Sequence number of the batch price update.
-    event BatchPriceFeedUpdate(uint16 chainId, uint64 sequenceNumber);
```
Comment (ali-bahjati) [inline target_chains/ethereum/sdk/solidity/IPythEvents.sol:22]: let's bump the sdk package version as major too.

**Review by ali-bahjati — COMMENTED** [review]
This is very nice. I left some inline comments to see whether we can remove even more code! I'll approve after getting your opinions about them.

File: target_chains/ethereum/contracts/forge-test/Pyth.t.sol (hunk 2 above)
Comment (Riateche) [reply target_chains/ethereum/contracts/forge-test/Pyth.t.sol:15]: There is some overlap between tests in `Pyth.t.sol` and `Pyth.WormholeMerkleAccumulator.t.sol`, but some tests are different.

There are similar tests for:

- testParsePriceFeedUpdatesWorks
- testParsePriceFeedUpdatesWorksWithRandomDistinctUpdatesInput
- testParsePriceFeedUpdatesRevertsIfUpdateFeeIsNotPaid
- testParsePriceFeedUpdatesRevertsIfPriceIdNotIncluded
- testParsePriceFeedUpdateRevertsIfPricesOutOfTimeRange
- testParsePriceFeedUpdatesLatestPriceIfNecessary

And these are the tests that are only in `Pyth.t.sol`:

- testParsePriceFeedUpdatesWorksWithOverlappingWithinTimeRangeUpdates
- testParsePriceFeedUpdatesWorksWithOverlappingMixedTimeRangeUpdates
- testParsePriceFeedUpdatesRevertsIfUpdateVAAIsInvalid
- testParsePriceFeedUpdatesRevertsIfUpdateSourceChainIsInvalid

Should I remove the duplicated tests? (I would still prefer to keep `testParsePriceFeedUpdatesWorks` just to make it easier to debug the `setUp()` if it ever breaks.)

File: target_chains/ethereum/contracts/forge-test/VerificationExperiments.t.sol (hunk 1 above)
Comment (Riateche) [reply target_chains/ethereum/contracts/forge-test/VerificationExperiments.t.sol]: I removed `PriceAttestation` and migrated `VerificationExperiments` to use `PriceFeedMessage` instead.

**Pull request merged**